pkg/bpf/map_linux_test.go

```diff
@@ -87,16 +113,270 @@ func (s *BPFPrivilegedTestSuite) TestOpen(c *C) {
 	// existingMap is the same as testMap. Opening should succeed.
 	existingMap := NewMap("cilium_test",
 		MapTypeHash,
-		4,
-		4,
+		int(unsafe.Sizeof(TestKey{})),
+		int(unsafe.Sizeof(TestValue{})),
 		maxEntries,
 		BPF_F_NO_PREALLOC,
 		0,
-		nil)
+		DumpParserFunc).WithCache()
 	err = existingMap.Open()
 	c.Check(err, IsNil)      // Avoid assert to ensure Close() is called below.
 	err = existingMap.Open() // Reopen should be no-op.
 	c.Check(err, IsNil)
 	err = existingMap.Close()
 	c.Assert(err, IsNil)
 }
+
+func (s *BPFPrivilegedTestSuite) TestOpenMap(c *C) {
+	openedMap, err := OpenMap("cilium_test_no_exist")
+	c.Check(err, Not(IsNil))
+	c.Check(openedMap, IsNil)
+
+	openedMap, err = OpenMap("cilium_test")
+	c.Check(err, IsNil)
+}
+
+func (s *BPFPrivilegedTestSuite) TestOpenOrCreate(c *C) {
+	// existingMap is the same as testMap. OpenOrCreate should skip recreation.
+	existingMap := NewMap("cilium_test",
+		MapTypeHash,
+		int(unsafe.Sizeof(TestKey{})),
+		int(unsafe.Sizeof(TestValue{})),
+		maxEntries,
+		BPF_F_NO_PREALLOC,
+		0,
+		DumpParserFunc).WithCache()
+	isNew, err := existingMap.OpenOrCreate()
+	c.Check(err, IsNil) // Avoid assert to ensure Close() is called below.
+	c.Check(isNew, Equals, false)
+
+	// preallocMap unsets BPF_F_NO_PREALLOC. OpenOrCreate should recreate map.
+	EnableMapPreAllocation() // prealloc on/off is controllable in HASH map case.
+	preallocMap := NewMap("cilium_test",
+		MapTypeHash,
+		int(unsafe.Sizeof(TestKey{})),
+		int(unsafe.Sizeof(TestValue{})),
+		maxEntries,
+		0,
+		0,
+		DumpParserFunc).WithCache()
+	// not to mess up emitted test results, checking isNew is sufficient.
+	logLevel := logging.DefaultLogger.GetLevel()
+	logging.DefaultLogger.SetLevel(logrus.ErrorLevel)
```

Comment:
This is really useful during test development, but do you think it's necessary in the tree?

If we want this everywhere, it can go into `runTests()`; otherwise we could remove them.

Reply:
Makes sense, will remove from the tree. Thanks.

Reply:
Ah no, it does not. I thought `DisableMapPreAllocation` could be out of the tree.
Setting log level to ErrorLevel during the whole test is acceptable?

Reply:
Sure. We don't usually have explicit changing of log level in the tests, but error sounds fine to me.

Reply:
@lkpdn Are you still planning on removing the setting of log level? Thx

Reply:
I was 50/50 about this but considering that nothing ensures that we'll never miss some important error message in these objCheck() call sites so removed it.